packages/x-date-pickers-pro/src/DateRangeCalendar/timezone.DateRangeCalendar.test.tsx

```diff
@@ -0,0 +1,41 @@
+import * as React from 'react';
+import { expect } from 'chai';
+import { screen, fireEvent } from '@mui-internal/test-utils';
+import { describeAdapters } from 'test/utils/pickers';
+import { DateRangeCalendar } from './DateRangeCalendar';
+
+describe('<DateRangeCalendar /> - Timezone', () => {
+  describeAdapters('Timezone prop', DateRangeCalendar, ({ adapter, render }) => {
+    if (!adapter.isTimezoneCompatible) {
+      return;
+    }
+
+    it('should correctly render month days when timezone changes', () => {
+      function DateCalendarWithControlledTimezone() {
+        const [timezone, setTimezone] = React.useState('Europe/Paris');
+        return (
+          <React.Fragment>
+            <DateRangeCalendar timezone={timezone} calendars={1} />
+            <button onClick={() => setTimezone('America/New_York')}>Switch timezone</button>
+          </React.Fragment>
+        );
+      }
+      render(<DateCalendarWithControlledTimezone />);
+
+      expect(
+        screen.getAllByRole('gridcell', {
+          name: (_, element) => element.nodeName === 'BUTTON',
+        }).length,
+      ).to.equal(30);
```

Comment:
I think the correct assertion should be no of grid cell of type button before and after the timezone change should be same. (I assume for month of february and month having 31 days it might fail)

and more accurate way of doing this will be before the the timezone change create the map of index of date in gridcell and compare it after the timezone change

Reply:
Thanks for noting. 🙏 
I've seen your tests and do agree that they are more "complete", but IMHO, I don't see the benefit of having a more complex-to-read test with marginally stricter assertions. The main thing that we need to assert here is that the same amount of days have been rendered after a timezone switch. 🤔 
I do value the benefit of simple tests, which you can glance over and understand in seconds. 🙈 
What do you think?

Reply:
I totally agree simple is better. But there might be chance in month of july and August also december 2024 and January 2025 when consecutive months have equal number of dates then this might not cover those edge cases.

Reply:
> But there might be chance in month of july and August also december 2024 and January 2025 when consecutive months have equal number of dates then this might not cover those edge cases.

Why do you think like that? 🤔 
The test is set to use `1` calendar for simplicity and ("current") time is mocked to `2022-06-15`.

Maybe you are talking about a separate issue (https://github.com/mui/mui-x/issues/10584) with AdapterDayjs and DST?
That's a separate topic that I'm looking into. 😉

Reply:
No I am not talking about DST. although I missed the date mocking part. I assume it would render the calendar of current. 

With date mocking your test will work. 

Sorry for missing date mocking part.